### ISSUE TITLE
Run clang format to format all C++ files

### ISSUE THREAD

**Issue by 4ian** [4ian/GD#456]
I attend to run clang-format to automatically reformat all C++ files, providing a cleaner codebase style. I will also enable faster coding as you can just code and call the "format document" option in an IDE like Visual Studio Code.
(It's already done using Prettier for JavaScript).

@Lizard-13 I'll do this once you're latest PR about skeleton is merged to minimize conflicts!

**Comment by 4ian** [comment]
I'll probably use WebKit style guide (see https://github.com/motine/cppstylelineup) but open to other suggestions

**Comment by 4ian** [comment]
Sure no problem! Take your time for checking, make sure that it's as good as you want and then let's merge this.
Do you think you'll be able to make an example? It will also be useful for me to regularly check that nothing is broken :)

Let me know! :)

**Comment by Lizard-13** [comment]
Sure, I can give that weird dragon with multiple deformable tails flames and flying polygons :D

**Comment by 4ian** [comment]
Would be awesome 😄
